app/main/views/frameworks.py

```diff
@@ -332,7 +324,17 @@ def download_agreement_file(framework_slug, document_name):
         abort(404)
 
     agreements_bucket = s3.S3(current_app.config['DM_AGREEMENTS_BUCKET'])
-    path = get_agreement_document_path(framework_slug, current_user.supplier_id, document_name)
+
+    if document_name == 'signed-agreement':
```

Comment:
The point of this is that we specifically do *not* want to use `get_document_path` for signed and countersigned documents, because the full path is now stored in the database and we want to use that. But we still need the old behaviour for result letters and signed agreement files.

~~So I guess the alternative to doing it this way, with specific keys for signed and countersigned, would be to make separate routes for signed and countersigned that look at supplier framework...~~

Scratch that, yes maybe a better way would be to have the key from the `supplier_framework` passed as the `document_name` parameter, look in the supplier framework for the key, and if it isn't found fall back to `get_document_path`.

This would introduce some camelCase to our document URLs though. Can we live with that?